pkg/operator/controller/controller.go

```diff
@@ -112,58 +109,42 @@ func (r *reconciler) reconcile(request reconcile.Request) (reconcile.Result, err
 			// resource that got deleted async).
 			logrus.Infof("clusteringress %q not found and reconciliation will be skipped", request)
 		} else {
-			// Try again later.
-			logrus.Errorf("failed to get clusteringress %q: %v", request, err)
-			result.RequeueAfter = 10 * time.Second
+			errs = append(errs, fmt.Errorf("failed to get clusteringress %q: %v", request, err))
 		}
 		ingress = nil
 	}
 
 	caSecret, err := r.ensureRouterCACertificateSecret()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure CA secret: %v", err)
+		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
 	}
 
-	// Collect errors as we go.
-	errs := []error{}
-
-	// Only reconcile the ingress itself if it exists.
 	if ingress != nil {
 		dnsConfig := &configv1.DNS{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
-			logrus.Errorf("failed to get dns 'cluster': %v", err)
+			errs = append(errs, fmt.Errorf("failed to get dns 'cluster': %v", err))
 			dnsConfig = nil
 		}
 		infraConfig := &configv1.Infrastructure{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
-			logrus.Errorf("failed to get infrastructure 'cluster': %v", err)
+			errs = append(errs, fmt.Errorf("failed to get infrastructure 'cluster': %v", err))
 			infraConfig = nil
 		}
-		if dnsConfig == nil || infraConfig == nil {
-			// For now, if the cluster configs are unavailable, defer reconciliation
-			// because weaving conditionals everywhere to deal with various nil states
-			// is too complicated. It doesn't seem too risky to rely on the invariant
-			// of the cluster config being available.
```

Comment:
I've been an advocate of doing whatever we can with what we have, but based on Dan's feedback on https://github.com/openshift/cluster-ingress-operator/pull/130, I think Dan wants to keep this invariant for now and avoid doing all those `dnsConfig == nil` and `infraConfig == nil` checks deeper in the controller code.

Reply:
I think we discussed this in our stand up call, when dnsConfig/infraConfig is not present we are able to perform other useful stuff. During my test runs, I have seen dnsConfig not found in the first iterator of reconciliation, we did ensure other resources and the request was queued and in the second iteration, ensureDNS succeeded. 
As Dan pointed out earlier, I have been seeing very aggressive retries by the rate limiter. Probably we need to introduce our custom rate limiter to mitigate this issue.

Reply:
Looking at the latest changes, ingressConfig is used in too many places and it is getting ugly and hard to reason if these nil checks are done later in the methods when ever it is necessary. So I will revert this to the current scheme of returning early when any of the dns/infra/ingress configs are not present.